app/models/patient.rb

```diff
@@ -204,6 +204,16 @@ def recent_history_tracks
     versions.where(updated_at: 6.days.ago..)
   end
 
+  def all_versions(include_fulfillment)
+    all_versions = versions || []
+    all_versions += (external_pledges.includes(versions: [:item, :user]).map(&:versions).reduce(&:+) || [])
+    all_versions += (practical_supports.includes(versions: [:item, :user]).map(&:versions).reduce(&:+) || [])
+    if include_fulfillment
+      all_versions += (fulfillment.versions.includes(fulfillment.versions.count > 1 ? [:item, :user] : []) || [])
+    end
```

Comment:
This is an awful hack, and in fact so awful that it makes me wonder if we should take out bullet from tests. The story here is that for most tests there's only one version associated with fulfillment, and so bullet just sees one result and assumes there's never a risk of an n+1 query. We get around that here by conditionally running the include if there _are_ multiple which gets bullet to chill, but potentially a 'cure worse than the disease' kinda thing here.

Reply:
I don't fully follow - what's happening if we just run it flat? Bullet is throwing a false positive?

Reply:
yeah - bullet can't seem to tell that versions is a has_many relation, and throws false positives if there's only one version. pretty annoying!